crates/core/src/workflows/mod.rs

```diff
@@ -89,21 +78,160 @@ async fn try_make_local_validation_package(
     }
 }
 
-/// Gets hold of the validation package for the given entry.
-/// First tries to create it locally and if that fails will try to get the
-/// validation package from the source.
+/// Gets hold of the validation package for the given entry by trying several different methods.
 async fn validation_package(
     entry_with_header: &EntryWithHeader,
     context: Arc<Context>,
 ) -> Result<Option<ValidationPackage>, HolochainError> {
-    // 1. Try to construct it locally:
-    if let Ok(package) =
-        try_make_local_validation_package(&entry_with_header, context.clone()).await
+    // 0. Call into the DNA to get the validation package definition for this entry
+    // e.g. what data is needed to validate it (chain, entry, headers, etc)
+    let entry = &entry_with_header.entry;
+    let validation_package_definition = get_validation_package_definition(entry, context.clone())
+        .and_then(|callback_result| match callback_result {
+        CallbackResult::Fail(error_string) => Err(HolochainError::ErrorGeneric(error_string)),
+        CallbackResult::ValidationPackageDefinition(def) => Ok(def),
+        CallbackResult::NotImplemented(reason) => Err(HolochainError::ErrorGeneric(format!(
+            "ValidationPackage callback not implemented for {:?} ({})",
+            entry.entry_type().clone(),
+            reason
+        ))),
+        _ => unreachable!(),
+    })?;
+
+    // 1. Try to construct it locally.
+    // This will work if the entry doesn't need a chain to validate or if this agent is the author:
+    log_debug!(
+        context,
+        "validation_package:{} - Trying to build locally",
+        entry_with_header.entry.address()
+    );
+    if let Ok(package) = try_make_local_validation_package(
+        &entry_with_header,
+        &validation_package_definition,
+        context.clone(),
+    )
+    .await
+    {
+        log_debug!(
+            context,
+            "validation_package:{} - Successfully built locally",
+            entry_with_header.entry.address()
+        );
+        return Ok(Some(package));
+    }
+
+    // 2. Try and get it from the author
+    log_debug!(
+        context,
+        "validation_package:{} - Could not build locally. Trying to retrieve from author",
+        entry_with_header.entry.address()
+    );
+
+    // this will block until the network goes down! It needs to timeout
```

Comment:
Is this comment true?  If so then this is a real problem.

Reply:
Ah nope this was the issue I was having that is fixed now

Reply:
The comment is still there FYI